src/sentry/sentry_metrics/consumers/indexer/multiprocess.py

```diff
@@ -68,14 +72,24 @@ def poll(self) -> None:
             self.__commit_function(self.__produced_message_offsets)
             self.__produced_message_offsets = {}
 
-    def submit(self, message: Message[Union[KafkaPayload, FilteredPayload]]) -> None:
+    def submit(self, message: Message[Union[KafkaPayload, FilteredPayload, None]]) -> None:
         if isinstance(message.payload, FilteredPayload):
             # FilteredPayload will not be commited, this may cause the the indexer to consume
             # and produce invalid message to the DLQ twice if the last messages it consume
             # are invalid and is then shutdown. But it will never produce valid messages
             # twice to snuba
             # TODO: Use the arroyo producer which handles FilteredPayload elegantly
             return
+
+        ((partition, offset),) = message.committable.items()
```

Comment:
if committable contains more than one partition/offset are you discarding the others?